### PR TITLE
Fixed all server errors caused by form submission

### PR DESCRIPTION
**Pull request by mattwatsoncodes**
There were a number of php errors in the log, this update adds a few escapes and fallbacks to ensure that these are not generated.

### CODE REVIEW

File: gfeloqua.class.php [hunk 1]
```diff
@@ -1682,13 +1685,17 @@ function retry_failed_submissions(){
 	 * @return array        Entries
 	 */
 	function get_failed_entries( $forms = array() ){
+
+		if ( empty( $forms ) ) {
+			return;
```
Comment (solepixel) [inline gfeloqua.class.php]: I think if we return an empty array here, that would keep from having to check if the returned value is an array. The expected response of this function should be an array.

File: includes/helpers.php [hunk 2]
```diff
@@ -5,3 +5,20 @@ function ctype_alnum( $text ){
...
+	 * Update Entry meta
+	 *
+	 * Allows a meta to be altered before posting to Eloqua.
+	 *
+	 * @param  string $key  The key of the metadata.
+	 * @param  string $lead The value of the metadata.
+	 * @param  object $form The form that the metadata belongs to.
+	 * @return string       Returns the value of the entry meta.
+	 */
+	function update_entry_meta( $key, $lead, $form ) {
```
Comment (solepixel) [inline includes/helpers.php:21]: Can you explain why `update_entry_meta` wouldn't exist?

**Comment by solepixel** [comment]
@mwtsn I wasn't aware these errors were showing up, but it has been awhile since I've used/worked with this plugin on a live site. If you feel these are necessary, I don't have any problems with most of them. I am a bit concerned about adding the `update_entry_meta` function without falling back on another GF function. Did GF change this function or remove it? Why are we having to re-create it? And why does it not do anything? Sorry for my ignorance, but I haven't had time to fully review and test everything so I'm just looking for a bit of clarity on some of these changes. Thanks!

**Comment by mattwatsoncodes** [comment]
Hi @solepixel,

No worries, you make some very good points with your comments and I'll address those this week.

With regards to `update_entry_meta` this function does not exist in the latest build of Gravity Forms (Trust me I've done several searches for it), so to stop the error I just created the function if it didn't already exist. 

I'll be perfectly honest I didn't dig into the code enough to figure out what your calls to that function did, but stopping the error let my data submit to Eloqua perfectly.

I hope that makes sense.

File: includes/helpers.php (hunk 2 above)
Comment (solepixel) [reply includes/helpers.php:21]: @mwtsn I did a bit more research and it looks like the latest version uses the prefered Gravity Form function `gform_update_meta`. Are you finding calls to `update_entry_meta` in the plugin somewhere? If not, this is probably not necessary.

Comment (mattwatsoncodes) [reply includes/helpers.php:21]: @solepixel yes there is a reference in gravityforms-eloqua/gfeloqua.class.php on line 1625:

`$meta[ GFELOQUA_OPT_PREFIX . 'success'] = array(
			'label' => __( 'Sent to Eloqua?', 'gfeloqua' ),
			'is_numeric' => false,
			'update_entry_meta_callback' => 'update_entry_meta',
			'is_default_column' => false
		);`

This is in the function `add_success_meta`

Comment (mattwatsoncodes) [reply includes/helpers.php:21]: @solepixel the function that calls that logic `add_success_meta` is hooked into the `form_update_meta` action:

`add_action( 'gform_entry_meta', array( $this, 'add_success_meta' ), 10, 2 );`

File: gfeloqua.class.php (hunk 1 above)
Comment (mattwatsoncodes) [reply gfeloqua.class.php]: My latest commit address this (0b14e20f508d97e2eaf808213366dc73aa2c3358)

File: includes/helpers.php (hunk 2 above)
Comment (mattwatsoncodes) [reply includes/helpers.php:21]: @solepixel looking further into the API documentation https://www.gravityhelp.com/documentation/article/gform_entry_meta/ if does seem you need to define your own custom function. 

Judging by the context of the call, I would say that the `update_entry_meta` function that has been defined in the helper file should return `true`.

Let me know if you agree.

**Pull request closed without merging**